frontend/libs/studio-content-library/src/ContentLibrary/LibraryBody/pages/CodeList/CodeLists/CodeLists.tsx

```diff
@@ -1,43 +1,69 @@
 import React from 'react';
 import type { CodeListWithMetadata } from '../CodeList';
 import { Accordion } from '@digdir/designsystemet-react';
-import { StudioCodeListEditor } from '@studio/components';
-import type { CodeList as StudioComponentsCodeList, CodeListEditorTexts } from '@studio/components';
-import { useOptionListEditorTexts } from '../hooks/useCodeListEditorTexts';
+import type { CodeList as StudioComponentsCodeList } from '@studio/components';
+import { EditCodeList } from './EditCodeList/EditCodeList';
+import { useTranslation } from 'react-i18next';
 
 export type CodeListsProps = {
   codeLists: CodeListWithMetadata[];
+  onUpdateCodeListId: (codeListId: string, newCodeListId: string) => void;
   onUpdateCodeList: (updatedCodeList: CodeListWithMetadata) => void;
+  codeListInEditMode: string | undefined;
+  codeListNames: string[];
 };
 
-export function CodeLists({ codeLists, onUpdateCodeList }: CodeListsProps) {
+export function CodeLists({
+  codeLists,
+  onUpdateCodeListId,
+  onUpdateCodeList,
+  codeListInEditMode,
+  codeListNames,
+}: CodeListsProps) {
   return codeLists.map((codeList) => (
-    <CodeList key={codeList.title} codeList={codeList} onUpdateCodeList={onUpdateCodeList} />
+    <CodeList
+      key={codeList.title}
+      codeList={codeList}
+      onUpdateCodeListId={onUpdateCodeListId}
+      onUpdateCodeList={onUpdateCodeList}
+      codeListInEditMode={codeListInEditMode}
+      codeListNames={codeListNames}
+    />
   ));
 }
 
 type CodeListProps = {
   codeList: CodeListWithMetadata;
+  onUpdateCodeListId: (codeListId: string, newCodeListId: string) => void;
   onUpdateCodeList: (updatedCodeList: CodeListWithMetadata) => void;
+  codeListInEditMode: string | undefined;
+  codeListNames: string[];
 };
 
-function CodeList({ codeList, onUpdateCodeList }: CodeListProps) {
-  const editorTexts: CodeListEditorTexts = useOptionListEditorTexts();
-
-  const handleUpdateCodeList = (updatedCodeList: StudioComponentsCodeList): void => {
-    const updatedCodeListWithMetadata = updateCodeListWithMetadata(codeList, updatedCodeList);
-    onUpdateCodeList(updatedCodeListWithMetadata);
-  };
+function CodeList({
+  codeList,
+  onUpdateCodeListId,
+  onUpdateCodeList,
+  codeListInEditMode,
+  codeListNames,
+}: CodeListProps) {
+  const { t } = useTranslation();
 
```

Comment:
Changes in this file are related to the suggestion in `EditCodeList.tsx`
```suggestion
  const invalidCodeListNames = codeListNames.filter(
    (codeListName) => codeListName !== codeListInEditMode,
  );
```